sympy/sets/ordinals.py

```diff
@@ -0,0 +1,208 @@
+    def convert(cls, integer_value):
+        return Ordinal([OmegaPower(0, integer_value)])
+
+    def __eq__(self, other):
+        if not isinstance(other, Ordinal):
+            try:
+                other = Ordinal.convert(other)
+            except TypeError:
+                return NotImplemented
+        return self.args == other.args
+
+    def __lt__(self, other):
+        if not isinstance(other, Ordinal):
+            try:
+                other = Ordinal.convert(other)
+            except TypeError:
+                return NotImplemented
+
+        for term_self, term_other in zip(self.args, other.args):
+            if term_self != term_other:
+                return term_self < term_other
+        return len(self.args) < len(other.args)
+
+    def __le__(self, other):
+        return (self == other or self < other)
+
+    def __gt__(self, other):
+        return not self <= other
+
+    def __ge__(self, other):
+        return not self < other
+
+    def __str__(self):
+        net_str = ""
+        plus_count = 0
+        for i in self.args:
+            if plus_count:
+                net_str += " + "
+            if i.mult == 0:
+                continue
+            elif i.exp == 0:
+                net_str += str(i.mult)
+            elif i.exp == 1:
+                if i.mult == 1:
+                    net_str += 'w'
+                else:
+                    net_str += 'w*%s'%i.mult
+            else :
+                if i.mult==1:
+                    net_str += 'w**%s'%i.exp
+                else:
+                    net_str += '{w**%s'%i.exp+'}*%s'%i.mult
+            plus_count += 1
+        return(net_str)
+
+    __repr__ = __str__
+
+    def __add__(self, other):
+
+        if isinstance(other, int):
+            new_terms = list(self.args)
+            if other < 0:
+                raise ValueError("can only add ordinals and positive integers")
+
+            elif new_terms[-1].exp == 0:
+                new_terms[-1] = OmegaPower(0, other+new_terms[-1].mult)
+                return Ordinal(new_terms)
+            else:
+                new_terms.append(OmegaPower(0, other))
+                return Ordinal(new_terms)
+        else:
+            a_terms = list(self.args)
+            b_terms = list(other.args)
+            power_self = [i.exp for i in a_terms]
+            power_other = [i.exp for i in b_terms]
+            b1 = power_other[0]
+            r = self._insert_index(power_self, power_other[0])
+            if not b1 in power_self:
+                a_terms.insert(r,OmegaPower(b1, 0))
+            else:
+                r-=1
+            net = a_terms[:r]
+            term2 = [b_terms[0].exp, b_terms[0].mult+a_terms[r].mult]
+            if term2:
+                net.append(OmegaPower(*term2))
+            term3 = (b_terms[1:])
+            if term3:
+                net.append(*term3)
+            return(Ordinal(net))
+
+    def __radd__(self, other):
+        if type(other) is int and other > 0:
```

Comment:
using `isinstance(other, (int, Integer))` is better. In general: don't use `type`, always prefer `isinstance` check. See [1].

.. [1] https://stackoverflow.com/a/1549854/3396379

Reply:
Or maybe rather `if not isinstance(other, Ordinal)`, then conversion is tried.